Framework/Algorithms/src/WienerSmooth.cpp

```diff
@@ -166,19 +164,34 @@ WienerSmooth::smoothSingleSpectrum(API::MatrixWorkspace_sptr inputWS,
     // add a fake value to the end to make size even
     inputWS = copyInput(inputWS, wsIndex);
     wsIndex = 0;
-    auto &X = inputWS->dataX(wsIndex);
-    auto &Y = inputWS->dataY(wsIndex);
-    auto &E = inputWS->dataE(wsIndex);
-    double dx = X[dataSize - 1] - X[dataSize - 2];
-    X.push_back(X.back() + dx);
-    Y.push_back(Y.back());
-    E.push_back(E.back());
+
+    auto &x = inputWS->x(wsIndex);
+    auto lastX = x.back();
+    auto dx = x[dataSize - 1] - x[dataSize - 2];
+
+    auto &y = inputWS->y(wsIndex);
+    size_t newSize = y.size() + 1;
+
+    auto histogram = inputWS->histogram(wsIndex);
+    histogram.resize(newSize);
+
+    auto &xResized = histogram.mutableX();
+    auto &yResized = histogram.mutableY();
+    auto &eResized = histogram.mutableE();
+    auto sizeXResized = xResized.size();
+    auto sizeYResized = yResized.size();
+    auto sizeEResized = eResized.size();
+
+    xResized[sizeXResized - 1] = lastX + dx;
+    yResized[sizeYResized - 1] = yResized[sizeYResized - 2];
+    eResized[sizeEResized - 1] = eResized[sizeEResized - 2];
+    inputWS->setHistogram(wsIndex, histogram);
```

Comment:
Your approach is cleaner, but `Histogram::size` is private(see [here](https://github.com/mantidproject/mantid/blob/master/Framework/HistogramData/inc/MantidHistogramData/Histogram.h#L202L206)) . I assume that this is unintentional. If so, i will make it public and use your implementation.